core/src/avm2/globals/xml.rs

```diff
@@ -81,6 +81,42 @@ pub fn name<'gc>(
     }
 }
 
+pub fn set_name<'gc>(
+    activation: &mut Activation<'_, 'gc>,
+    this: Object<'gc>,
+    args: &[Value<'gc>],
+) -> Result<Value<'gc>, Error<'gc>> {
+    let xml = this.as_xml_object().unwrap();
+    let node = xml.node();
+
+    let new_name = args.get_value(0);
```

Comment:
now that we have `isXMLName`, we should use it here and throw an exception if the name is illegal, right? (we can do this in a follow-up if you prefer)

Reply:
I would prefer to do it in this PR.